test/config/utility.cc

```diff
@@ -1397,15 +1397,22 @@ void ConfigHelper::initializeTls(
   } else {
     if (options.client_with_intermediate_cert_) {
       validation_context->add_verify_certificate_hash(TEST_CLIENT2_CERT_HASH);
-      std::string cert_yaml = R"EOF(
+      std::string cert_yaml;
+      if (options.trust_root_only_) {
+        cert_yaml = R"EOF(
         trusted_ca:
-          filename: "{{ test_rundir }}/test/config/integration/certs/intermediate_ca_cert_chain.pem"
+          filename: "{{ test_rundir }}/test/config/integration/certs/cacert.pem"
       )EOF";
-      if (options.max_verify_depth_) {
-        cert_yaml += R"EOF(
-        max_verify_depth: 1
+      } else {
+        cert_yaml = R"EOF(
+        trusted_ca:
+          filename: "{{ test_rundir }}/test/config/integration/certs/intermediate_partial_ca_cert_chain.pem"
       )EOF";
       }
+      if (options.max_verify_depth_.has_value()) {
+        cert_yaml += R"EOF(
+        max_verify_depth: )EOF" + std::to_string(options.max_verify_depth_.value());
```

Comment:
in case people make the same mistake in the future :) you can move this after the line 1416, then set max_verify_depth by `validation_context.set_max_verify_depth()`